### PR TITLE
prepare docs for 4.1 release

### LINKED ISSUE #63: Timestamps in message headers - implemented but not specified
The code adds a timestamp to every message header, under the key 'date'. The [messaging spec](http://jupyter-client.readthedocs.org/en/latest/messaging.html) does not describe this, and extensions to the spec belong in metadata, not the header. So either it should be mentioned in the docs, or the code shouldn't do it.

### LINKED ISSUE #73: Refresh docs to be more user and developer friendly
In preparation for 4.1, I going to do a basic refresh of the docs.
- [x] Check Sphinx builds cleanly without warnings
- [x] Add captions to table of contents to improve readability
- [x] Add a more prominent statement about this containing protocol information

### PR DESCRIPTION
**Pull request by minrk**
- note date header (closes #63)
- update docs, including changelog and a few fixes (closes #73)

### CODE REVIEW

**Comment by minrk** [comment]
I'll merge this by the end of today, and release 4.1.

File: docs/messaging.rst [hunk 1]
```diff
@@ -129,6 +131,13 @@ A message is defined by the following four-dictionary structure::
 
    ``version`` key added to the header.
 
+.. versionchanged:: 5.1
+
+    ``date`` in the header was accidentally ommitted from the spec prior to 5.1,
```
Comment (willingc) [inline docs/messaging.rst]: Typo ommitted -> omitted

**Comment by willingc** [comment]
@minrk Overall, this looks good. 

I did get some build errors thrown by Sphinx that was preventing the blocking_client from building via autodoc so the added text would not render locally. FYI I applied your branch on master before the build.

```
sphinx-build -b html -d _build/doctrees   . _build/html
Running Sphinx v1.3.1
loading pickled environment... failed: 'str' object has no attribute '__dict__'
loading intersphinx inventory from http://ipython.org/ipython-doc/dev/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 9 source files that are out of date
updating environment: 9 added, 0 changed, 0 removed
reading sources... [ 11%] api/client
reading sources... [ 22%] api/index
reading sources... [ 33%] api/kernelspec
reading sources... [ 44%] api/manager
reading sources... [ 55%] changelog
reading sources... [ 66%] index
reading sources... [ 77%] kernels
reading sources... [ 88%] messaging
reading sources... [100%] wrapperkernels

/Users/carol/Projects/carol-jupyter/jupyter_client/docs/api/client.rst:27: WARNING: autodoc: failed to import method 'KernelClient.comm_info' from module 'jupyter_client'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/util/inspect.py", line 109, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'KernelClient' has no attribute 'comm_info'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 392, in import_object
    obj = self.get_attr(obj, part)
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 288, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/util/inspect.py", line 115, in safe_getattr
    raise AttributeError(name)
AttributeError: comm_info
/Users/carol/Projects/carol-jupyter/jupyter_client/docs/api/manager.rst:30: WARNING: autodoc: failed to import method 'KernelManager.blocking_client' from module 'jupyter_client'; the following exception was raised:
Traceback (most recent call last):
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/util/inspect.py", line 109, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'KernelManager' has no attribute 'blocking_client'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 392, in import_object
    obj = self.get_attr(obj, part)
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/ext/autodoc.py", line 288, in get_attr
    return safe_getattr(obj, name, *defargs)
  File "/Users/carol/Projects/carol-jupyter/jupvenv/lib/python3.5/site-packages/sphinx/util/inspect.py", line 115, in safe_getattr
    raise AttributeError(name)
AttributeError: blocking_client
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [ 11%] api/client
writing output... [ 22%] api/index
writing output... [ 33%] api/kernelspec
writing output... [ 44%] api/manager
writing output... [ 55%] changelog
writing output... [ 66%] index
writing output... [ 77%] kernels
writing output... [ 88%] messaging
writing output... [100%] wrapperkernels

generating indices... genindex py-modindex
writing additional pages... search
copying images... [100%] figs/frontend-kernel.png

copying static files... done
copying extra files... done
dumping search index in English (code: en) ... done
dumping object inventory... done
build succeeded, 2 warnings.

Build finished. The HTML pages are in _build/html.
```

**Comment by minrk** [comment]
Thanks! Typos fixed. I verified that it builds correctly on readthedocs: http://minrk-jupyter-client.readthedocs.org/en/prepare-4.1. I think you had 4.0 of jupyter-client installed, which doesn't have those new methods. I see the same thing when I haven't yet installed the master version.

**Comment by willingc** [comment]
@minrk Glad it was just the version level. Figured I would flag it just in case it was something else :sunrise:

**Pull request merged**